main.js

```diff
@@ -0,0 +1,187 @@
+var posts = [];
+var currentIndex;
+var currentPost;  
+
+//adds new posts based on inputs in main-form and renders them-also stores inputs in posts array with unique id
+var newPost = function () {
+  var $userText = $('#post-text').val();
+  var $userName = $('#name').val(); 
+
+  var post = {
+    id: uuidv4(), 
+    post: $userText,
+    name: $userName,
+    comment: []
+  };
+
+  posts.push(post); 
+
+  //each new post is the last post in the posts array
+  $postText = posts[posts.length-1].post;  
+  $postName = posts[posts.length-1].name; 
+  $postId = posts[posts.length-1].id;
+
+  renderPost();
+ 
+  //adds uniqu id to each post-el div 
+  $post = $('.post-el'); 
+  $post.last().attr('id', $postId); 
+};
+
+var renderPost = function () {
+  //$('#submit').unbind(); 
+   $('.posts').append('<div class="post-el"><p class="post">' + '<button type="button" class="btn btn-link edit-button">edit</button>' + '<button type="button" class="btn btn-link remove-button">remove</button>' + '<button type="button" class="btn btn-link comment-button">comment</button><button type="button" class="btn btn-link comments-button">comments</button>'  + '<span class="post-text">' + $postText + '</span>' + '<span class="name-text">' + ' - Posted By: ' + $postName + '</span>' + '<hr></p></div>'); 
+
+  $('form :input').val(''); //resets main form  
+};
+
+var removeItemsFromPosts = function (array1) {
+  for (let i = 0; i < array1.length; i++) {
+    const element = array1[i]; 
+
+    if ($currentPostId === element.id) {
+      currentIndex = array1.indexOf(element); 
+    };
+  };
+  array1.splice(currentIndex, 1); 
+};
+
+var removePost = function () {
+  $currentPostId = $(this).parents('.post-el').attr('id'); 
+  removeItemsFromPosts(posts); 
+
+  $('.posts').append($('#comments-form')); //moves comment-form so it doesn't get deleted
+  $(this).parents('.post-el').remove(); 
+};
+
+var openComment = function () {
+  $(this).parent('.post').append($('#comments-form')); //comments form opens beneath the comment selected
```

Comment:
Interesting! No one else has done this is (defining something in the DOM then utilizing it). I'm curious if there are any downsides to doing this.

Reply:
I'm not sure if I'm understanding what you mean by defining something in the DOM. and then using it. I know I wrote it, but I didn't know that's what I was doing (lol). I thought I was just appending my comment form to the comment button that is added when someone makes a new post. Can you explain further?